src/client/Microsoft.Identity.Client/ManagedIdentity/V2/InterprocessLock.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.Identity.Client.PlatformsCommon.Shared;
+
+namespace Microsoft.Identity.Client.ManagedIdentity.V2
+{
+    /// <summary>
+    /// Executes paramref name="action"/ under a cross-process, per-alias mutex.
+    /// We attempt 2 namespaces, in order:
+    /// 1) <c>Global\</c> — preferred so we dedupe across all sessions on the machine
+    ///    (e.g., service + user session). This can be denied by OS policy or missing
+    ///    SeCreateGlobalPrivilege in some contexts.
+    /// 2) <c>Local\</c> — fallback to still dedupe within the current session when
+    ///    <c>Global\</c> is not permitted.
+    /// Using both ensures we never throw (persistence is best-effort) while getting
+    /// machine-wide dedupe when allowed and session-local dedupe otherwise.
+    /// Notes:
+    /// - The mutex name is derived from <c>alias</c> (= cacheKey) via SHA-256 hex (truncated)
+    ///   to avoid invalid characters / length issues.
+    /// - On non-Windows runtimes the Global/Local prefixes are treated as part of the name;
+    ///   behavior remains correct but dedupe scope is platform-defined.
+    /// - Abandoned mutexes are treated as acquired to avoid blocking after a crash.
+    /// </summary>
+    internal static class InterprocessLock
+    {
+        // Prefer Global\ for cross-session dedupe; fall back to Local\
+        // if ACLs block Global\ to remain non-throwing.
+        public static bool TryWithAliasLock(
+            string alias,
+            TimeSpan timeout,
+            Action action,
+            Action<string> logVerbose = null)
+        {
+            var globalName = GetMutexNameForAlias(alias, preferGlobal: true);
+            var localName = GetMutexNameForAlias(alias, preferGlobal: false);
+
+            bool TryScope(string name, out bool unauthorized)
```

Comment:
nit: add xml docs to this function so we can know what it does without looking at the rest of its code

Reply:
added